griptape/utils/structure_visualizer.py

```diff
@@ -17,6 +19,8 @@ class StructureVisualizer:
     structure: Structure = field()
     header: str = field(default="graph TD;", kw_only=True)
     build_node_id: Callable[[BaseTask], str] = field(default=lambda task: task.id.title(), kw_only=True)
+    query_params: dict[str, str] = field(factory=dict, kw_only=True)
+    base_url: str = field(default="https://mermaid.ink", kw_only=True)
```

Comment:
should these just be passed directly to the `to_url()` method? they arent really necessary for the state of the object, only used for generating a url

Reply:
That argument could be made for most of the parameters here, right? If anything I'd say that _structure_ belongs on `to_url()` with the idea being that you'd create one `StructureVisualizer` for many Structures. But the work required to change that in a non-breaking way feels like it outweighs the value of a silly util.

Reply:
eh maybe. there could be other `to_` methods that need the other stuff, but not these. it also seems reasonable to generate different URLs with different params without updating the property every time.